### PR TITLE
Change `==` operator for `XPCRepresentable` from public to internal

### LINKED ISSUE #8: `func ==(lhs: XPCRepresentable, rhs: XPCRepresentable) -> Bool` override implementations of Swift Standard Library
I inspected implementation of `matchPattern(_: withSyntaxKinds:)` in [File+SwiftLint.swift](https://github.com/realm/SwiftLint/blob/fe75393/Source%2FSwiftLintFramework%2FExtensions%2FFile%2BSwiftLint.swift#L70-L76) and noticed following:
```swift
            return kindsInRange.count == syntaxKinds.count &&
                zip(kindsInRange, syntaxKinds).filter({ $0.0 != $0.1 }).isEmpty
```
I wondered why does not it use following 🤔:
```swift
            return kindsInRange == syntaxKinds
```

`SyntaxKind` is enum with raw `String` and conforms `Equatable` by default. So, `Array<SyntaxKind>` can use Swift Standard Library's implementation:
```swift
/// Returns true if these arrays contain the same elements.
@warn_unused_result
public func ==<Element : Equatable>(lhs: [Element], rhs: [Element]) -> Bool
```

But, I tested SwiftLint using `kindsInRange == syntaxKinds` in that and got test fails. :astonished: 

Then, I found that SwiftXPC has following implementation:
```swift
/// Enable comparison of XPCRepresentable objects.
public func ==(lhs: XPCRepresentable, rhs: XPCRepresentable) -> Bool {
```

On that test, [default case in switch](https://github.com/jpsim/SwiftXPC/blob/261fbd45d2b0786790ca1f6c90de94c569a3b543/SwiftXPC%2FXPCType.swift#L108-L111) of that was used.
```swift
    default:
        // Should never happen because we've checked all XPCRepresentable types
        return false
    }
```
I think this is an issue that should be resolved.

### PR DESCRIPTION
**Pull request by norio-nomura**
Add `XPCRepresentable.isEqualTo(_:)` instead of that.
Fix #8

### CODE REVIEW

File: SwiftXPC/XPCType.swift [hunk 1]
```diff
@@ -53,12 +62,12 @@ public typealias XPCDictionary = [String: XPCRepresentable]
 // MARK: Equatable
 
 /// Enable comparison of XPCRepresentable objects.
-public func !=(lhs: XPCRepresentable, rhs: XPCRepresentable) -> Bool {
+internal func !=(lhs: XPCRepresentable, rhs: XPCRepresentable) -> Bool {
```
Comment (jpsim) [inline SwiftXPC/XPCType.swift]: I don't think this is ever used anymore

File: SwiftXPC/XPCType.swift [hunk 2]
```diff
@@ -53,12 +62,12 @@ public typealias XPCDictionary = [String: XPCRepresentable]
...
     return !(lhs == rhs)
 }
 
 /// Enable comparison of XPCRepresentable objects.
-public func ==(lhs: XPCRepresentable, rhs: XPCRepresentable) -> Bool {
+internal func ==(lhs: XPCRepresentable, rhs: XPCRepresentable) -> Bool {
```
Comment (jpsim) [inline SwiftXPC/XPCType.swift]: why keep this if only used in `isEqualTo`?

**Comment by norio-nomura** [comment]
Thanks for reviews!
Updated.

**Comment by jpsim** [comment]
Sweet!

**Comment by norio-nomura** [comment]
Thanks! :pray:

**Pull request merged**